Sources/Core/Views/TextField/TextField.swift

```diff
@@ -0,0 +1,273 @@
+//
+//  TextField.swift
+//  PovioKit
+//
+//  Created by Arlind Dushi on 01/20/2022.
+//  Copyright © 2022 Povio Inc. All rights reserved.
+//
+
+import UIKit
+
+public protocol RuleValidatable {
+  var error: String { get }
+  func validate(_ input: String?) -> Bool
+}
+
+public class TextField: UIView {
+  private let stackView = UIStackView()
+  private let titleLabel = PaddingLabel()
+  private let valueContainerView = UIView()
+  private let valueTextField = UITextField()
+  private let errorLabel = UILabel()
+  private var rule: RuleValidatable?
+  private var textFieldHeightAnchor: NSLayoutConstraint?
+
+  weak public var delegate: UITextFieldDelegate? {
+    didSet { valueTextField.delegate = delegate }
+  }
+  
+  public init(with rule: RuleValidatable? = .none) {
+    super.init(frame: .zero)
+    self.rule = rule
+    setupViews()
+  }
+  
+  required init?(coder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  @discardableResult
+  override public func becomeFirstResponder() -> Bool {
+    valueTextField.becomeFirstResponder()
+  }
+  
+  @discardableResult
+  override public func resignFirstResponder() -> Bool {
+    valueTextField.resignFirstResponder()
+  }
+  
+  override public var isFirstResponder: Bool {
+    valueTextField.isFirstResponder
+  }
+  
+  public var inputFieldHeight: CGFloat = 44 {
+    didSet { textFieldHeightAnchor?.constant = inputFieldHeight }
+  }
+  
+  public var inputFieldCornerRadius: CGFloat = 10 {
```

Comment:
This is duplicated with _textFieldCornerRadius_
I guess the name textFieldCornerRadius is more appropriate in this case to stay consistent with the rest of the properties...